app/src/app.vue

```diff
@@ -85,6 +86,41 @@ export default defineComponent({
 			{ immediate: true }
 		);
 
+		async function setAppLanguage() {
+			let lang = 'en-US'; // fallback language, if no language is set
+			if (userStore.currentUser && userStore.currentUser.language) {
+				// if user has set it's own language, prefer user language
+				lang = userStore.currentUser.language;
+			} else if (serverStore.info?.project?.project_language) {
+				// otherwise use project language
+				lang = serverStore.info.project.project_language;
+			}
+			await setLanguage(lang);
+		}
+
+		watch(
+			() => appStore.hydrating,
+			async () => {
+				// initially set app language after hydration
+				if (!appStore.hydrating) await setAppLanguage();
```

Comment:
IIRC the app store hydrates after user login. The serverStore is hydrated _before_ the user logs in. Ideally, we set the correct language after the serverStore is hydrated, so we can show the login page in the correct language as well

Reply:
Yeah, you are right. I did quite some testing however and it needs both watches: To watch the server store hydration for setting the language of the login screen and in a second stage the app hydration to set the language when the user's language is known.

Reply:
> Yeah, you are right. I did quite some testing however and it needs both watches: To watch the server store hydration for setting the language of the login screen and in a second stage the app hydration to set the language when the user's language is known.

Yeah. Just curious how it ties in with the existing `setLanguage` calls, and making sure we're not doing the same thing redundantly in multiple places:

https://github.com/directus/directus/blob/ef18cdc14470aea5d6e35d0f4c1ee91aba71d132/app/src/hydrate.ts#L78

https://github.com/directus/directus/blob/ef18cdc14470aea5d6e35d0f4c1ee91aba71d132/app/src/modules/users/routes/item.vue#L438-L449

Reply:
You are right, for app hydration setLanguage is called twice.

Would you prefer to have the initial setLanguage calls in hydrate.ts and server.ts (server store) or to handle it via watchers in app.vue?